modules/storages/app/common/storages/adapters/providers/share_point/share_point_registry.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Adapters
+    module Providers
+      module SharePoint
+        SharePointRegistry = Dry::Container::Namespace.new("share_point") do
```

Comment:
:yellow_circle: I just realize this now, and it is not yet too late, so let's have this conversation early :D

When visiting the Microsoft website the product seems to be called `SharePoint`, but in URLs they write `sharepoint`.

Obviously, this is inconsistent w.r.t. how we'd convert between pascal case and snake case.

That said, from a perspective of code readability, I think I'd vote for `sharepoint` in snake case and `Sharepoint` in pascal case. E.g. `SharepointRegistry`.

For example if a product name would be written in all caps, we'd also not add the `ALLCAPSPRODUCTRegistry`.

Thoughts from your side? Next time we talk about this, the number of files will be larger :D